packages/eslint-plugin-internal/src/rules/debug-namespace.ts

```diff
@@ -0,0 +1,66 @@
+import type { TSESTree } from '@typescript-eslint/utils';
+
+import { getStaticValue } from '@typescript-eslint/utils/ast-utils';
+
+import { createRule } from '../util';
+
+function filePathToNamespace(filePath: string) {
+  const matched =
+    /typescript-eslint[\\/]+packages[\\/]+([a-z-]+)[\\/]+(.+)/.exec(filePath);
```

Comment:
Probably better to just make this a path relative to this file? Eg `pkgsRoot = path.resolve('..', '..', '..'); relativePath = file.replace(pkgsRoot, '..');`

That way the rule doesn't rely on the folder name.

Reply:
Ah, those irksome to deal with in unit tests with explicit `filename`s. But +1 that this regex is unseemly. I switched to `.split(/packages[\\/]+/)` and added a bit more testing.